ToDoApp/ToDoApp/Respository/Respository.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Respository.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-22.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+protocol Repository {
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID
+    func read<T>() async -> Result<[T], Error> where T: Decodable
+}
+extension DatabaseReference: Repository {
+  // MARK: create a task
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID {
+        do {
+           try await self.child(value.id.uuidString).setValue(value.toDictionary)
```

Comment:
value to dictionary should be unwrapped before being passed to the setValue method. if there would be an error while encoding it, instead of providing an error the data would be destroyed